src/device/CrashDump.cpp

```
@@ -0,0 +1,364 @@
+        default:
+            throw std::runtime_error("Unsupported platform in crash dump: " + platform2string(platform));
+    }
+
+    return result;
+}
+
+CrashDump::~CrashDump() = default;
+
+std::vector<uint8_t> CrashDump::toBytes() const {
+    fs::path tempPath = platform::getTempPath();
+    fs::path tarPath = tempPath / "crashdump.tar";
```

Cppcheck (2.20.0)

[error] There is an unknown macro here somewhere. Configuration is required. If DEPTHAI_NLOHMANN_DEFINE_TYPE_INTRUSIVE is a macro then please configure it.

(unknownMacro)